### PR TITLE
Main method refactor

### PR DESCRIPTION
**Pull request by liamhebert**
Lots of changes here
- Updated message sending and receiving repo with new structure (`message` -> `internal-messaging`) 
- Wrote `main()` method which uses test data. Checks for messages every 5 seconds
  - Every 1 minute, a `REQUEST` message is sent and the GPS is polled and sent
  - Every 30 seconds, a `EMPTY` message (no contents) is sent, and the GPS waits
  - If neither of these thresholds are met, a `STANDBY` message is sent, and the GPS waits
 -  Every 2 minutes (will be 720 in production), if no `REQUEST` messages have been sent during that time, the GPS is polled.
- Changed methods that require `struct`'s to pass them by reference and not value for increased speed. 
- Moved message sending and receiving code into `GPSMessageInterface`
- Added include guards

**_NOTE_**: This says 69 files changed, it's really just 4 (`GPS.cpp`, `GPS.hpp`, `GPSMessageInterface.cpp` and `GPSMessageInterface.hpp`) The rest is all code from the OS repo that has been snapshotted into our repo

### CODE REVIEW

File: Source/source/GPS.cpp [hunk 1]
```diff
@@ -29,9 +27,10 @@ bool poll(string *message) {
 }
 
 //send in a NMEA string, get organized gps_data
-gps_data decode(string raw) { 
-	gps_data data = {};
+void decode(string raw) { 
+	data = {};
 	minmea_sentence_gga ggaFrame;
+	minmea_sentence_zda zdaFrame;
```
Comment (betheddy) [inline Source/source/GPS.cpp:32]: Is this used anywhere? should there be a minmea_parse_zda call below?

File: Source/source/GPS.cpp [hunk 2]
```diff
@@ -44,16 +43,10 @@ gps_data decode(string raw) {
...
 		}
 	}
-	struct gps_time decoded_time = decode_time_from_int(data.time);
-	printf("time: %s\nenc_time: %s\naltitude: %f\n(%f, %f)\nheight: %f\n\n",
-	data.time_stamp.c_str(), toStringTime(&decoded_time).c_str(), data.altitude, data.latitude, data.longitude, data.height);
-	
-	return data;
 }
 
 string toStringTime(struct minmea_time *time) {
```
Comment (betheddy) [inline Source/source/GPS.cpp]: Do we need both toStringTime methods?

File: Source/source/GPS.cpp (hunk 1 above)
Comment (liamhebert) [reply Source/source/GPS.cpp:32]: Ah this code was removed by accident. I'll re-implement that now

File: Source/source/GPS.cpp (hunk 2 above)
Comment (liamhebert) [reply Source/source/GPS.cpp]: Nice catch! I'll delete that now

File: Source/source/GPSMessageInterface.hpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+#include <math.h>
+#include <chrono>
+#include "Identifiers.h"
+
+using namespace std;
+//#include "MessageSenderInterface.h"
+
+bool send_message(gps_data* decoded_data);
+Message get_message();
+
```
Comment (betheddy) [inline Source/source/GPSMessageInterface.hpp]: You should declare the "decode_time_from_int" method here

**Review by betheddy — CHANGES_REQUESTED** [review]

File: Source/source/GPSMessageInterface.hpp (hunk 3 above)
Comment (liamhebert) [reply Source/source/GPSMessageInterface.hpp]: Fixed this

**Review by kaylee-kim — APPROVED** [review]

**Review by betheddy — APPROVED** [review]

**Review by jporterdal — APPROVED** [review]

**Pull request merged**